### PR TITLE
fix: point to ids column instead of index

### LINKED ISSUE #704: Bug in unit testing (possible upstream problem)
#### Problem description

Please explain:
* **what** you tried to achieve,
* **how** you went about it (referring to the code sample), and
* **why** the current behaviour is a problem and what output
  you expected instead.

`pytest` is failing because [`cobra.flux_analysis.single_cell_deletion`](https://github.com/opencobra/cobrapy/blob/d9d3bf99198ee0a93346bd860fce4cf217297937/src/cobra/flux_analysis/deletion.py#L236) is returning the gene ids as a column "ids" instead of as the index (as it is in its docstring). Maybe this is a problem on cobrapy or the expected behavior has changed and the docstring is outdated. In the latter case, I have a branch working here https://github.com/carrascomj/memote/tree/fix-test-essentiality

#### Code Sample

Create a [minimal, complete, verifiable example](https://stackoverflow.com/help/mcve).
Under the memote repository root:

```shell
python -m pytest tests/test_for_experimental/test_for_config.py
```

Traceback: https://github.com/opencobra/memote/pull/703/checks

### Context

<details>

```
System Information
==================
OS                  Linux
OS-release 5.8.12-arch1-1
Python             3.6.12

Package Versions
================
Jinja2                          2.11.2
click                            7.1.2
click-configfile                 0.2.3
click-log                        0.3.2
cobra                           0.19.0
cookiecutter                     1.7.2
depinfo                          1.5.4
equilibrator-api                0.1.26
future                          0.18.2
gitpython                        3.1.9
goodtables                       2.5.1
importlib-resources              3.0.0
lxml                             4.5.2
memote              0.11.1+20.gea06738
numpydoc                         1.1.0
pandas                           1.1.2
pip                             20.2.1
pylru                            1.2.0
pytest                           6.1.0
requests                        2.24.0
ruamel.yaml                    0.16.12
setuptools                      50.1.0
six                             1.15.0
sqlalchemy                      1.3.19
sympy                            1.6.2
travis-encrypt                   1.1.2
wheel                           0.35.1
```

</details>

### PR DESCRIPTION
**Pull request by carrascomj**
* [x] fix #704 
`pytest` is failing because an essentially test is extracting the wrong information from [`cobra.flux_analysis.single_cell_deletion`](https://github.com/opencobra/cobrapy/blob/d9d3bf99198ee0a93346bd860fce4cf217297937/src/cobra/flux_analysis/deletion.py#L236), whose behavior has recently changed. 
* [x] description of feature/fix
Change index to column access. An if statement was added to the list comprehension because it made the error more clear, but feel free to remove it.
* [x] tests added/passed
* [x] add an entry to the [next release](../HISTORY.rst)

### CODE REVIEW

File: src/memote/experimental/essentiality.py [hunk 1]
```diff
@@ -92,7 +93,9 @@ def evaluate(self, model):
             essen = single_gene_deletion(
                 model, gene_list=self.data["gene"], processes=1
             )
-        essen["gene"] = [list(g)[0] for g in essen.index]
+        essen["gene"] = [
+            list(g)[0] if isinstance(g, Iterable) else g for g in essen["ids"]
```
Comment (Midnighter) [inline src/memote/experimental/essentiality.py:97]: Aren't elements of `essen["ids"]` always sets?

**Review by Midnighter — COMMENTED** [review]
Huh, black formatting is pretty weird now. One question on your actual change, otherwise it looks good to go in.

File: src/memote/experimental/essentiality.py (hunk 1 above)
Comment (carrascomj) [reply src/memote/experimental/essentiality.py:97]: Right, that filter was just clarifying the error, but it is superfluous now.

**Review by Midnighter — APPROVED** [review]
LGTM

**Pull request merged**